diff --git a/include/ws_core/types/args.h b/include/ws_core/types/args.h
index cf36c00..dd7111b 100644
--- a/include/ws_core/types/args.h
+++ b/include/ws_core/types/args.h
@@ -4,9 +4,16 @@
 #include <map>
 #include <stdint.h>
 #include <ws_core/macros.h>
+#include <cstring>
 
 // Cool argument types
 
+struct compare_strings {
+    bool operator()(const char* a, const char* b) const {
+        return strcmp(a, b) < 0;
+    }
+};
+
 // Use an X macro (https://en.wikipedia.org/wiki/X_macro) for ease
 #define WS_CORE_ARG_TYPE(X) \
     X(FLAG)                 \
@@ -91,7 +98,7 @@ namespace WS_Core {
         // Probably shouldn't use a vector here
         extern std::vector<Arg> args;
         // Also probably shouldn't use a map here
-        extern std::map<const char*, Value> values;
+        extern std::map<const char*, Value, compare_strings> values;
 
         /// @brief Parse arguments
         /// @param argc Argument count
diff --git a/src/core/args.cpp b/src/core/args.cpp
index ae579a0..db641ae 100644
--- a/src/core/args.cpp
+++ b/src/core/args.cpp
@@ -50,7 +50,7 @@ namespace WS_Core::Args {
             .arg_long = "port",
             .arg_short = "p",
 
-            .arg_type = Type::PORT
+            .arg_type = PORT
         },
         Arg {
             .name = "Internal Port",
@@ -73,7 +73,7 @@ namespace WS_Core::Args {
             .arg_long = "port",
             .arg_short = "Ip",
 
-            .arg_type = Type::PORT
+            .arg_type = PORT
         },
         Arg {
             .name = "Version",
@@ -89,11 +89,10 @@ namespace WS_Core::Args {
 
             .arg_long = "version",
             .arg_short = "v",
-
-            .arg_type = TYPE::FLAG
+            .arg_type = FLAG
         }
     };
-    std::map<const char*, Value> values;
+    std::map<const char*, Value, compare_strings> values;
 }
 
 void str_to_lower(const char* o, char* n) {
@@ -109,7 +108,9 @@ bool str_to_bool(const char* str) {
     if(strcmp(lower, "on") == 1 ||
         strcmp(lower, "true") == 1) {
         return true;
-    } else if(strcmp(lower, "off") == 1 ||
+    }
+
+    if(strcmp(lower, "off") == 1 ||
         strcmp(lower, "false") == 1) {
         return false;
     }
@@ -204,14 +205,14 @@ bool Args::parse_args(int argc, char* const argv[]) {
     
     for(int i = 1; i < argc; i++) {
         if(find_arg(argv[i], &current_arg)) {
-            switch(Args::args[i].arg_type) {
-                case Args::Type::FLAG:
+            switch(args[i].arg_type) {
+                case FLAG:
                     if(i > argc - 1) {
                         // Check to see if it is a form of boolean
                         if(str_is_bool(argv[i + 1])) {
                             bool val = str_to_bool(argv[i + 1]);
 
-                            Args::set_value(
+                            set_value(
                                 current_arg.internal_name,
                                 val
                             );
@@ -225,67 +226,67 @@ bool Args::parse_args(int argc, char* const argv[]) {
                     // default true, without specifying
                     // false will do nothing
 
-                    Args::set_value(
+                    set_value(
                         current_arg.internal_name,
                         true
                     );
                     break;
-                case Args::Type::INTEGER:
+                case INTEGER:
                     if(i == argc - 1) {
                         // Invalid arguments
                         return false;
                     }
-                    Args::set_value(
+                    set_value(
                         current_arg.internal_name,
                         std::atoi(argv[i + 1]),
                         false
                     );
                     i++;
                     break;
-                case Args::Type::STRING:
+                case STRING:
                     if(i == argc - 1) {
                         // Invalid arguments
                         return false;
                     }
-                    Args::set_value(
+                    set_value(
                         current_arg.internal_name,
                         argv[i + 1],
                         false
                     );
                     i++;
                     break;
-                case Args::Type::FLOAT:
+                case FLOAT:
                     if(i == argc - 1) {
                         // Invalid arguments
                         return false;
                     }
-                    Args::set_value(
+                    set_value(
                         current_arg.internal_name,
                         std::atof(argv[i + 1]),
                         false
                     );
                     i++;
                     break;
-                case Args::Type::DOUBLE:
+                case DOUBLE:
                     if(i == argc - 1) {
                         // Invalid arguments
                         return false;
                     }
-                    Args::set_value(
+                    set_value(
                         current_arg.internal_name,
                         std::atof(argv[i + 1]),
                         false
                     );
                     i++;
                     break;
-                case Args::Type::PATH:
+                case PATH:
                     // Need to check that the path is valid
                     // even if the file/directory it is pointed
                     // at does not exist
 
                     /// @todo Path implementation
                     break;
-                case Args::Type::PORT:
+                case PORT:
                     // This one is a little special
                     // It needs checking to ensure
                     // that it is within valid port
@@ -297,7 +298,7 @@ bool Args::parse_args(int argc, char* const argv[]) {
                         // Invalid arguments
                         return false;
                     }
-                    int port = std::atoi(argv[i + 1]);
+                    auto port = strtol(argv[i + 1], nullptr, 10);
 
                     // Do checks
                     if(port <= 255 && port >= 65535) {
@@ -319,60 +320,64 @@ bool Args::parse_args(int argc, char* const argv[]) {
             return false;
         }
     }
+
+    return true;
 }
 
-Args::Value Args::get_value(const char* key, Args::Value def) {
-    auto i = Args::values.find(key);
-    if(i == Args::values.end()) {
+Args::Value Args::get_value(const char* key, Value def) {
+    auto i = values.find(key);
+    if(i == values.end()) {
         def.using_default = true;
         return def;
-    } else {
-        Value val = Args::values[key];
-        val.using_default = false;
-        return val;
     }
+
+    Value val = values[key];
+    val.using_default = false;
+    return val;
 }
 
 Args::Value Args::get_value(const char* key) {
-    auto i = Args::values.find(key);
-    if(i == Args::values.end()) {
-        Args::Value val;
+    auto i = values.find(key);
+    if(i == values.end()) {
+        Value val;
         val.success = false;
         return val;
     }
+
+    return {};
 }
 
 void Args::set_value(const char* key, Arg::Default val, Type type) {
-    Args::values[key].value = val;
-    Args::values[key].type = type;
+    values[key].value = val;
+    values[key].type = type;
 }
 
 void Args::set_value(const char* key, bool flag) {
     Arg::Default val;
     val.flag = flag;
-    Args::set_value(key, val, Type::FLAG);
+    set_value(key, val, FLAG);
 }
 
 void Args::set_value(const char* key, int val, bool port) {
-    Arg::Default _val;
+    Arg::Default _val{};
     _val.integer = val;
-    Args::set_value(key, _val, port ? Type::PORT : Type::INTEGER);
+    set_value(key, _val, port ? PORT : INTEGER);
 }
 
 void Args::set_value(const char* key, const char* string, bool path) {
     Arg::Default val;
     strcpy(val.string, string);
-    Args::set_value(key, val, path ? Type::PATH : Type::STRING);
+    set_value(key, val, path ? PATH : STRING);
 }
 
 void Args::set_value(const char* key, float flt) {
     Arg::Default val;
     val.flt = flt;
-    Args::set_value(key, val, Type::FLOAT);
+    set_value(key, val, FLOAT);
 }
 
 void Args::set_value(const char* key, double dbl) {
     Arg::Default val;
     val.dbl = dbl;
-    Args::set_value(key, val, Type::DOUBLE);
+    set_value(key, val, DOUBLE);
 }
\ No newline at end of file
diff --git a/src/core/help.cpp b/src/core/help.cpp
index 957789b..1d7d064 100644
--- a/src/core/help.cpp
+++ b/src/core/help.cpp
@@ -1,3 +1,4 @@
+#include <iostream>
 #include <ws_core/ws.h>
 #include <ws_core/types/args.h>
 #include <ws_core/help.h>
@@ -5,10 +6,6 @@
 using namespace WS_Core;
 
 void Help::print_usage_info() {
-    printf(
-        "WeatherStack Core - Version %d.%d.%d - Usage information\n",
-        VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH
-    );
-
-    printf("%d total available arguments\n", Args::args.size());
+    printf(std::format("WeatherStack Core - Version {}.{}.{} - Usage information\n", VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH).c_str());
+    printf(std::format("{} total available arguments\n", Args::args.size()).c_str());
 }
\ No newline at end of file
